src/cmd/run.js

```diff
@@ -117,6 +122,18 @@ export default async function run(
   const customPrefs = pref;
   const manifestData = await getValidatedManifest(sourceDir);
 
+  const profileDir = firefoxProfile || chromiumProfile;
+
+  if (profileCreateNew && profileDir) {
+    const isDir = fs.existsSync(profileDir);
+    if (isDir) {
+      throw new WebExtError(`Directory ${profileDir} already exists.`);
```

Comment:
@akhilpanchal I did re-think a bit more about this and this part of the approach that I suggested in my comment https://github.com/mozilla/web-ext/issues/2058#issuecomment-717184560 is actually not covering the use case describe in the [issue description](https://github.com/mozilla/web-ext/issues/2058#issue-729774986) in a nice way, because we will throw an error if the directory exist (while the workaround launch script in the issue description is not throwing in that case).

My apologies for missing to notice this when I wrote the initial guidance comment in the issue.

I think that relaxing this and just log a message (we may opt for a `log.info` if we want the user to see it even if web-ext run isn't in verbose mode) if the directory already exist may be actually reasonable (and it would better cover the enhancement request)

To better match the expected behavior we may also renamed the option as `--profile-create-if-missing`.

Reply:
I completely agree with you about not throwing and just logging an info message that tells the user that the directory already exists. Will update this behavior and also update the option name.